dev/PowerNotifications/PowerNotifications.h

```diff
@@ -7,8 +7,71 @@
 #include <powersetting.h>
 #include <Microsoft.Windows.System.Power.PowerManager.g.h>
 #include <frameworkudk\PowerNotificationsPal.h>
-#include <wil\resource.h>
-#include <wil\result_macros.h>
+#include <..\WindowsAppSDK_Insights\WindowsAppRuntimeInsights.h>
+
+class PowerNotifications : public wil::TraceLoggingProvider
+{
+    IMPLEMENT_TRACELOGGING_CLASS(PowerNotifications, "Microsoft.WindowsAppSDK.System.PowerNotifications",
+        // {a1b12e2c-12d9-564e-2ea1-2894ffcc7cc5}
+        (0xa1b12e2c, 0x12d9, 0x564e, 0x2e, 0xa1, 0x28, 0x94, 0xff, 0xcc, 0x7c, 0xc5));
+
+    public:
+
+        DEFINE_COMPLIANT_TELEMETRY_EVENT_STRING(RegisterEvent, PDT_ProductAndServiceUsage, value);
```

Comment:
Minor: consider changing from "value" to something more descriptive (e.g. functionName, dataName, etc.)

Reply:
Also, isn't capturing the name explicitly redundant if the event is happening in the context of an activity that's going to have the same information? If RegisterEvent fires with "EnergySaverStatus" as its parameter, and that happens under an activity named "EnergySaverStatusRegisterActivity", the parameter may not be doing much. Unless there's some reason that the activity name is harder to use when processing the telemetry for some reason (most of my experience is with deciphering raw events in ETL files, rather than using the full telemetry pipeline.

Reply:
Changed. 
For the other comment, I am using Activities to log API calls and not all of the Register Activities will result in a registration on the FrameworkUDK side. So I added the events to explicitly log it and differentiate. Hope this makes sense.

Reply:
> not all of the Register Activities will result in a registration on the FrameworkUDK side
I'm not questioning whether the event is necessary, I'm asking whether the *parameter* of the event adds any information.
The key question for whether the parameter is redundant is "is it possible for AddCallback/RemoveCallback to be called outside the context of one of these entrypoint activities?". If not, then the event should always have an activity tag that embeds the same information that's in the parameter.

Reply:
No longer applicable as we now only use two events.